### ISSUE TITLE
RobolectricGradleTestRunner fails to find androidManifest.xml for library module with Android Instrumentation Tests

### ISSUE THREAD

**Issue by stephenhand** [robolectric/robolectric#1925]
I tried to add JUnit tests to a library module (using Robolectric 3.0) and found that the runner fails because it cannot find a manifest in the intermediate files.

The problem, looking at the getAppManifest code is that it checks if a 'manifests' directory exists in intermediates, and if it does, it assumes that intermediates/manifests/full/{buildflavour}/androidManifest.xml must exist and sets the manifest path to that, otherwise fall back to setting it to a file in the bundles directory.

My library module has an intermediates/manifests directory but no 'full' directory, only an 'androidTest' directory. Presumably this is because this module produces no APK for production but does produce one for running android tests in the emulator.

It looks like the directory checking code just needs to be a little more robust to cover this case, I can provide a pull request with a proposed fix if needed.

**Comment by Ali-Rezaei** [comment]
I faced with exactly same issue and force to create folders manually. Because as explained for 'com.android.application' we have full folder but not for 'com.android.library'

**Comment by nenick** [comment]
I have created some project variants with robolectric support. So I can say that testing library modules work. Maybe you find a missing part. https://github.com/nenick/AndroidStudioAndRobolectric

Which robolectric version do you use (snapshot or rc-X)?

**Comment by Ali-Rezaei** [comment]
Mine is 3.0-rc3. I will look into that. Thanks for update.

**Comment by jaredsburrows** [comment]
@stephenhand Can you show use your setup? Maybe screenshot your IDE? I have multi-flavor apps that all run fine on Robolectric 3.0.

**Comment by jaredsburrows** [comment]
@erd Can we close this?

**Comment by stephenhand** [comment]
@jaredsburrows sorry I forgot to follow this up, I will try to get some more info along the lines you asked for when I'm next working on the project in question on Monday, but I worked around it by implementing a slightly modified runner so if you'd rather close it in the meantime that's ok with me.

**Comment by stephenhand** [comment]
@jaredsburrows On further investigation this is only an issue if you don't do a clean of the solution in between running 'Android Instrumentation Tests' for the project and 'Unit Tests'. If you clean after you switch it gets rid of the 'manifests' folder that sends the runner down the wrong path (this gets created to run the androidTest ), so this would make it a very minor issue.

**Comment by lnanek** [comment]
Running into this problem a lot with Gradle plugin 2.2.0 and Robolectric 3.1 even without a clean task right before. Maybe 2.2.0 changes where the manifest XMLs get saved to.

I can't upgrade to Robolectic 3.1.1 because it has some sort of JAR depending on AAR error in the shadows dependency. 

I was able to fix the Robolectric 3.1 bug of not generating the manifests and looking for them in the wrong place by adding this to my build.gradle for the library though:
````
task copyFilesDebug(type: Copy) {
    description = 'Fix Robolectric looks wrong place for AndroidManifest.xml by copying file...'
    from 'build/intermediates/manifests/aapt/debug'
    into 'build/intermediates/manifests/full/debug'
    include 'AndroidManifest.xml'
}

task copyFilesRelease(type: Copy) {
    description = 'Fix Robolectric looks wrong place for AndroidManifest.xml by copying file...'
    from 'build/intermediates/manifests/aapt/release'
    into 'build/intermediates/manifests/full/release'
    include 'AndroidManifest.xml'
}

project.afterEvaluate {
    test.dependsOn assemble
    test.dependsOn copyFilesDebug
    test.dependsOn copyFilesRelease
}
````

Posting solution here since this page comes up when you Google all the Robolectic errors about not finding the manifest XML.
